cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -62,8 +66,21 @@ protected boolean removeEldestEntry(Map.Entry<String, AnomalyState> eldest) {
       });
     }
     _selfHealingEnabled = selfHealingEnabled;
+    _ongoingSelfHealingAnomaly = new AtomicReference<>();
```

Comment:
Would you clarify why we use an `AtomicReference` rather than `volatile` here?

Reply:
since there are two thread can update this value, I feel `volatile` will have consistency issue. I change back to `synchronized`  due to reason above.

Reply:
1. Excerpt from https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/package-summary.html
```
The memory effects for accesses and updates of atomics generally follow the rules for volatiles, as stated in The Java Language Specification (17.4 Memory Model):

get has the memory effects of reading a volatile variable.
set has the memory effects of writing (assigning) a volatile variable.
```
-- i.e. the typical benefit of `AtomicReference` is CAS. `volatile` has lower overhead; hence, should be preferred against `AtomicReference` when possible.

2. Changing to `synchronized` (from `volatile` or `AtomicReference`) to handle multiple writers seems correct solution to me.